data/crac/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/FlowCnecImplTest.java

```diff
@@ -535,4 +535,50 @@ public void testEqualsAndHashCode() {
         assertEquals(cnec1.hashCode(), cnec1.hashCode());
         assertNotEquals(cnec1.hashCode(), cnec2.hashCode());
     }
+
+    @Test
+    public void testIsConnected() {
+        Network network = NetworkImportsUtil.import12NodesNetwork();
+        NetworkImportsUtil.addDanglingLine(network);
+
+        // Branch
+        FlowCnec cnec1 = crac.newFlowCnec()
+            .withId("cnec-1-id")
+            .withNetworkElement("BBE1AA1  BBE2AA1  1")
+            .withInstant(Instant.PREVENTIVE)
+            .newThreshold().withUnit(Unit.MEGAWATT).withMax(1000.).withRule(BranchThresholdRule.ON_LEFT_SIDE).add()
+            .add();
+        assertTrue(cnec1.isConnected(network));
+
+        network.getBranch("BBE1AA1  BBE2AA1  1").getTerminal1().disconnect();
+        assertFalse(cnec1.isConnected(network));
+
+        network.getBranch("BBE1AA1  BBE2AA1  1").getTerminal1().connect();
+        network.getBranch("BBE1AA1  BBE2AA1  1").getTerminal2().disconnect();
```

Comment:
je ne comprends pas pourquoi tu retestes disconnect, tu ne veux pas plutôt tester connect(), assertTrue(isconnected()) ?

Reply:
I'm testing that if terminal 1 OR terminal 2 is disconnected, then overall state is disconnected

Reply:
ah, my bad, je n'avais pas bien lu